src/transformers/tokenization_transfo_xl.py

```diff
@@ -271,6 +271,33 @@ def add_symbol(self, sym):
             self.idx2sym.append(sym)
             self.sym2idx[sym] = len(self.idx2sym) - 1
 
+    def move_added_token(self, token: str, target_idx: int):
+        """
+        Moves an added token to a specific position in the vocab.
+        This method should be used when resizing an embedding layer other than the last one in the `AdaptiveEmbedding`
+        in order to move the token in the tokenizer from the default position (at the very end) to the desired one.
+
+        Args:
+            token: The token to move to a specific position in the vocab.
+            target_idx: The position where the token should be moved to.
+        """
+        assert token in self.added_tokens_encoder, "Token which should be moved has to be an added token"
+        assert token not in self.idx2sym, "Token which should be moved is already in vocab"
+
+        # Insert sym into vocab
+        self.idx2sym.insert(target_idx, token)
+        self.sym2idx[token] = target_idx
+
+        # Shift following indices in sym2idx
+        for idx in range(target_idx + 1, len(self.idx2sym)):
+            current_sym = self.idx2sym[idx]
+            self.sym2idx[current_sym] = idx
+
+        # Delete token from added_tokens
+        old_index = self.added_tokens_encoder[token]
+        del self.added_tokens_decoder[old_index]
+        del self.added_tokens_encoder[token]
```

Comment:
Actually, I just had a discussion with @TevenLeScao and I think it would be better to not delete the token from the dicts `self.added_tokens_decoder` and `self.added_tokens_encoder` because we have these dicts to keep track which tokens have been added manually. Also pinging @LysandreJik here.

Could you instead maybe just change the `range( ...)` of your loop so that **any** token can be shifted for transfo-xl and not just the last token? 

(If I understand the functionality correctly, here...also pinging @TevenLeScao so that he can maybe elaborate)

Reply:
Ok yeah, I understand that. So you would prefer that I edit the index in `self.added_tokens_decoder` and `self.added_tokens_encoder` and do not insert the token into the original vocab? 

But I have to shift the tokens with the indices `[target_idx : len(vocab)]` in order to make the "virtual" position-change of the token possible, AFAIK. And this would result in an empty field at `self.idx2sym[target_idx]` which could cause some other issues? E.g. the vocab_size would be wrong due to the aforementioned empty field.

> Could you instead maybe just change the `range( ...)` of your loop so that any token can be shifted for transfo-xl and not just the last token?

Sorry, I did not understand that. From my point of view you can shift already **any** token?

Reply:
Can't we just add insert the token in the original vocab and also leave it in `self.added_tokens_encoder`?

In any case, you can't shift any token here; there's the assertion to ensure it is in `self.added_tokens_encoder`, and I am not sure it would work to move a token down the index rather than up the index.

Reply:
> Can't we just add insert the token in the original vocab and also leave it in `self.added_tokens_encoder`?

I tried that before, but then `len(tokenizer)` is +1 higher than it should be. I guess it's because the original vocab and the added_tokens are summed up.

Reply:
Yep I see. The way I see it, this is the right solution atm, and we'd need to take a look at the TransfoXL tokenizer later after the next refactor to do better. I'd say LGTM for now @patrickvonplaten @LysandreJik !